src/engine/surface_load.c

```diff
@@ -55,12 +53,17 @@ u32 gTotalStaticSurfaceData;
  * Allocate the part of the surface node pool to contain a surface node.
  */
 static struct SurfaceNode *alloc_surface_node(u32 dynamic) {
-    struct SurfaceNode **poolEnd = (struct SurfaceNode **)(dynamic ? &gDynamicSurfacePoolEnd : &gCurrStaticSurfacePoolEnd);
+    struct SurfaceNode *node;
```

Comment:
this code is the main thing i am worried about. are we absolutely sure main_pool_alloc has zero overhead? i can't find the definition so cant check myself. this would call thousands of times on level load and also hundreds of times per frame for dynamic surfaces so this will be bad if it has any measurable RAM or CPU overhead

Reply:
That is the key point of this PR. I want `main_pool_alloc` to have zero overhead period. It allows us to use it for all sorts of various applications that require extra memory during level runtime like allocating `Object` or new static surfaces.

Reply:
I see, that sounds good to me then